core/core.js

```diff
@@ -1,411 +1,123 @@
-+---------------------------------------------------------------+
-|                        RECORDS (*MAX_RECORDS)               ...
-+---------------------------------------------------------------+
- */
 "use strict";
 
 ((window) => {
-  const MAX_RECORDS = 100;
-  const INDEX_NUM_RECORDS = 0;
-  const INDEX_NUM_SHIFTED_OFF = 1;
-  const INDEX_HEAD = 2;
-  const INDEX_OFFSETS = 3;
-  const INDEX_RECORDS = INDEX_OFFSETS + 2 * MAX_RECORDS;
-  const HEAD_INIT = 4 * INDEX_RECORDS;
-
   // Available on start due to bindings.
   const core = window.Deno.core;
   const { recv, send } = core;
 
-  ////////////////////////////////////////////////////////////////////////////////////////////
-  ///////////////////////////////////////// Dispatch /////////////////////////////////////////
-  ////////////////////////////////////////////////////////////////////////////////////////////
-
-  const dispatch = send;
-  const dispatchByName = (opName, control, ...zeroCopy) =>
-    dispatch(opsCache[opName], control, ...zeroCopy);
-
-  ////////////////////////////////////////////////////////////////////////////////////////////
-  //////////////////////////////////// Shared array buffer ///////////////////////////////////
-  ////////////////////////////////////////////////////////////////////////////////////////////
-
-  let sharedBytes;
-  let shared32;
-
   let opsCache = {};
+  const errorMap = {};
 
   function init() {
-    const shared = core.shared;
-    assert(shared.byteLength > 0);
-    assert(sharedBytes == null);
-    assert(shared32 == null);
-    sharedBytes = new Uint8Array(shared);
-    shared32 = new Int32Array(shared);
-
-    asyncHandlers = [];
-    // Callers should not call core.recv, use setAsyncHandler.
     recv(handleAsyncMsgFromRust);
   }
 
   function ops() {
     // op id 0 is a special value to retrieve the map of registered ops.
-    const opsMapBytes = send(0);
-    const opsMapJson = String.fromCharCode.apply(null, opsMapBytes);
-    opsCache = JSON.parse(opsMapJson);
-    return { ...opsCache };
-  }
-
-  function assert(cond) {
-    if (!cond) {
-      throw Error("assert");
-    }
-  }
-
-  function reset() {
-    shared32[INDEX_NUM_RECORDS] = 0;
-    shared32[INDEX_NUM_SHIFTED_OFF] = 0;
-    shared32[INDEX_HEAD] = HEAD_INIT;
-  }
-
-  function head() {
-    return shared32[INDEX_HEAD];
-  }
-
-  function numRecords() {
-    return shared32[INDEX_NUM_RECORDS];
-  }
-
-  function size() {
-    return shared32[INDEX_NUM_RECORDS] - shared32[INDEX_NUM_SHIFTED_OFF];
-  }
-
-  function setMeta(index, end, opId) {
-    shared32[INDEX_OFFSETS + 2 * index] = end;
-    shared32[INDEX_OFFSETS + 2 * index + 1] = opId;
-  }
-
-  function getMeta(index) {
-    if (index >= numRecords()) {
-      return null;
-    }
-    const buf = shared32[INDEX_OFFSETS + 2 * index];
-    const opId = shared32[INDEX_OFFSETS + 2 * index + 1];
-    return [opId, buf];
+    return opsCache = Object.freeze(Object.fromEntries(send(0)));
```

Comment:
Nitpick, please split assignment and return into two separate lines for readability.